src/WebJobs.Script.WebHost/Security/SimpleWebTokenHelper.cs

```diff
@@ -94,7 +99,15 @@ public static bool TryValidateToken(string token, ISystemClock systemClock)
 
         public static bool ValidateToken(string token, ISystemClock systemClock)
         {
-            var data = Decrypt(GetWebSiteAuthEncryptionKey(), token);
+            // Use ContainerEncryptionKey if available else fallback to WebSiteAuthEncryptionKey.
+            // Until the container is specialized to a specific site WebSiteAuthEncryptionKey will not be available.
+            byte[] key;
+            if (!GetEncryptionKey(EnvironmentSettingNames.ContainerEncryptionKey, out key, false))
```

Comment:
After specialization, won't ContainerEncryptionKey still be set, so you never fall back? I.e. shouldn't your fallback logic be switched around to test for WebSiteAuthEncryptionKey first, which will be missing until specialization?

Reply:
Not falling back to WebAuthEncryptionKey is ok. ContainerKey already guarantees the request was internal. 
When ContainerKey is not available (non linux container mode) the other key will be used for internal requests. So either order should be okay.